### PR TITLE
Nativizing

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

**Comment by jfalcou** [comment]
Looks good but I think you need to make a quick pass to un-natify the trivial cases like for the unsigned abs and some others.

File: include/boost/simd/arch/common/generic/function/divides.hpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ namespace boost { namespace simd { namespace ext
                           , bd::generic_<bd::unspecified_<T>>
                           )
   {
-    BOOST_FORCEINLINE T operator()(const fast_tag &, T const& a, T const& b
+     BOOST_FORCEINLINE T operator()(const fast_tag &, T const& a, T const& b
```
Comment (jfalcou) [inline include/boost/simd/arch/common/generic/function/divides.hpp:34]: noisy change

File: include/boost/simd/arch/common/scalar/function/remnearbyint.hpp [hunk 2]
```diff
@@ -34,8 +34,8 @@ namespace boost { namespace simd { namespace ext
                           , (typename A0)
                           , bd::cpu_
                           , bs::tag::nearbyint_
-                          , bd::generic_< bd::int_<A0> >
-                          , bd::generic_< bd::int_<A0> >
+                          , bd::scalar_< bd::int_<A0> >
```
Comment (jfalcou) [inline include/boost/simd/arch/common/scalar/function/remnearbyint.hpp:36]: do we have a SIMD counterpart or what was the reason of this change ?

File: include/boost/simd/arch/common/simd/function/abs.hpp [hunk 3]
```diff
@@ -42,8 +43,9 @@ namespace boost { namespace simd { namespace ext
 
   //------------------------------------------------------------------------------------------------
   // abs on unsigned values is identity
-  BOOST_DISPATCH_OVERLOAD ( abs_
+  BOOST_DISPATCH_OVERLOAD_IF ( abs_
```
Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/abs.hpp]: this one is probably unneeded

File: include/boost/simd/arch/common/simd/function/enumerate.hpp [hunk 4]
```diff
@@ -19,7 +19,7 @@ namespace boost { namespace simd { namespace ext
 {
   // -----------------------------------------------------------------------------------------------
   // enumerate<T>()
-  BOOST_DISPATCH_OVERLOAD ( enumerate_
+  BOOST_DISPATCH_OVERLOAD( enumerate_
```
Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/enumerate.hpp:22]: noisy change

File: test/function/simd/modf.cpp [hunk 5]
```diff
@@ -23,18 +23,21 @@ void test(Env& $)
 
   using p_t = bs::pack<T, N>;
 
-  T a1[N], m[N], e[N];
+  T a1[N], ent[N], frac[N];
```
Comment (jfalcou) [inline test/function/simd/modf.cpp:25]: isn't this unrelated to the topic of nativizing ?

File: include/boost/simd/arch/common/simd/function/unary_plus.hpp [hunk 6]
```diff
@@ -19,8 +19,9 @@ namespace boost { namespace simd { namespace ext
   namespace bs = boost::simd;
   namespace bd = boost::dispatch;
 
-  BOOST_DISPATCH_OVERLOAD(unary_plus_
+  BOOST_DISPATCH_OVERLOAD_IF(unary_plus_
```
Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/unary_plus.hpp]: This one is identity , thus maybe unneeded

File: test/function/simd/modf.cpp (hunk 5 above)
Comment (jtlap) [inline test/function/simd/modf.cpp:25]: Yes and no. After nativizing a compilation pb appeared for modf and I find it still had the old by ref parameters...
So in the run...

Comment (jfalcou) [inline test/function/simd/modf.cpp:25]: Fine.

**Pull request merged**